### DISCUSSION TITLE
Julian Day calculation beyond 4713 BC.

### DISCUSSION THREAD

**Discussion by BikramAjay** [Stellarium/stellarium#1825] [General]
Can someone help me understand (or point to the suitable code file name) the Julian Day calculation implemented in Stellarium for the dates beyond January 1, 4713 BC. 
![Screenshot 2021-07-31 122417](https://user-images.githubusercontent.com/86772525/127746263-face606d-4cbf-4c1a-87b0-6deb1af19f08.jpg)
![Screenshot 2021-07-31 122349](https://user-images.githubusercontent.com/86772525/127746266-20467cb0-6533-45ef-a48b-93b843cc7e69.jpg)

"1. Julian day number (JDN)

The Julian day number associated with the solar day is the number assigned to a day in a continuous count of days beginning with the JULIAN DAY NUMBER 0 ASSIGNED TO THE DAY STARTING AT GREENWICH MEAN NOON ON 1 JANUARY 4713 BC, Julian proleptic calendar -4712."

https://www.iers.org/IERS/EN/Science/Recommendations/resolutionB1.html

**Comment by BikramAjay** [comment]
Ignore it. It's just negative counting.

**Reply by gzotti** [reply]
To find something in the source code, make a global search (grep -rin Julian), and you will find it faster than asking here and requiring our own time.

**Reply by BikramAjay** [reply]
Thanks. Will do.

**Reply by BikramAjay** [reply]
Here is an issue I am facing understanding Julian Date calculation as implemented in Stellarium. 

The first two screenshots are from the Stelutils.cpp. The second one is from the book "Numerical Recipies in C". 
I coded both and looked at the value of jul variable for both implementations at the point as shown by the red arrow. 

Let us assume
Jul (from Numerical recipies in C) =jul_A
jul (from the code in Steluntils.cpp) = jul_B
jul (as displayed by Stellarium GUI) = jul_C

Here is what I found

when input year value is >0, jul_A=jul_B=jul_C.

when input year is <0 then jul_A=jul_B≠ jul_C

eg When input date June 1, 1

jul_A=jul_B=jul_C= 1721575.0000000000
and when input date June 1, -1
jul_A=jul_B = 1721210.0000000000 and jul_C =1720843 

If anyone can throw some light on this discrepancy, it will be very helpful. 
Thanks in advance. 

![Screenshot 2021-08-01 225910](https://user-images.githubusercontent.com/86772525/127799021-3fcff822-b830-473f-8262-0e1b645e8fea.jpg)
![Screenshot 2021-08-01 230020](https://user-images.githubusercontent.com/86772525/127799029-eda742d8-79a9-4772-8e5e-9d6c61fb59aa.jpg)

**Reply by BikramAjay** [reply]
![Screenshot 2021-08-01 225415](https://user-images.githubusercontent.com/86772525/127800563-e1adec77-f351-42a3-8f64-9b6bd1e00284.jpg)

**Reply by gzotti** [reply]
* User Guide: Appendix F.8
* https://github.com/Stellarium/stellarium/wiki/FAQ#there-is-no-year-0-or-bc-dates-are-a-year-out

**Reply by BikramAjay** [reply]
Ah!!  I knew this but somehow it never came to my mind that will affect Julian date calculation as well. #Learning

Thanks!
